test/e2e/qat/qatplugin_qatgen4_crypto.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package qat
+
+import (
+	"path/filepath"
+	"time"
+
+	"github.com/intel/intel-device-plugins-for-kubernetes/test/e2e/utils"
+	"github.com/onsi/ginkgo/v2"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/framework/kubectl"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+const (
+	qatPluginKustomizationYaml = "deployments/qat_plugin/overlays/e2e/kustomization.yaml"
+	opensslTestYaml            = "demo/openssl-qat-engine-cpa-sample-pod.yaml"
+)
+
+func init() {
+	ginkgo.Describe("QAT Gen4 plugin in Crypto mode", describeQatPluginCy)
+}
+
+func describeQatPluginCy() {
+	f := framework.NewDefaultFramework("qatplugincy")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+
+	kustomizationPath, err := utils.LocateRepoFile(qatPluginKustomizationYaml)
+	if err != nil {
+		framework.Failf("unable to locate %q: %v", qatPluginKustomizationYaml, err)
+	}
+
+	opensslTestYamlPath, err := utils.LocateRepoFile(opensslTestYaml)
+	if err != nil {
+		framework.Failf("unable to locate %q: %v", opensslTestYaml, err)
+	}
+
+	ginkgo.It("measures performance of QAT Cy Services", func() {
+		ginkgo.By("deploying QAT plugin in DPDK mode")
+		framework.RunKubectlOrDie(f.Namespace.Name, "--namespace", f.Namespace.Name, "apply", "-k", filepath.Dir(kustomizationPath))
```

Comment:
Maybe a stupid question, but why is the _f.Namespace.Name_ given twice?

Reply:
It's a good question. We seem to be using it consistently wrong and here it's just copy-paste from existing code. I think it can be dropped and fixed in other places in a separate PR.

Reply:
As the overall removal was merged, this should be changed similarly.